packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -0,0 +1,217 @@
+  const grid = useRef();
+
+  const gridRef = useCallback((node) => {
+    grid.current = node;
+    const focusedCell = grid.current.rows[focusedRow.current].cells[focusedCol.current];
+    focusedCell.tabIndex = 0;
+  }, []);
+
+  const handleKeyDown = (event) => {
+    const currentFocusedRow = event.target.parentElement.rowIndex;
+    const currentFocusedCol = event.target.cellIndex;
+
+    let nextRow = currentFocusedRow;
+    let nextCol = currentFocusedCol;
+
+    const key = event.keyCode;
+    switch (key) {
+      case KeyCode.KEY_UP:
+        nextRow -= 1;
+        break;
+      case KeyCode.KEY_DOWN:
+        nextRow += 1;
+        break;
+      case KeyCode.KEY_LEFT:
+        nextCol -= 1;
+        break;
+      case KeyCode.KEY_RIGHT:
+        nextCol += 1;
+        break;
+      case KeyCode.KEY_HOME:
+        nextCol = 0;
+        if (event.ctrlKey) {
+          nextRow = 1; // Assumption is that the first row is the column Heading.
+        }
+        break;
+      case KeyCode.KEY_END:
+        nextCol = props.rows[0].cells.length - 1;
+        if (event.ctrlKey) {
+          nextRow = props.rows.length;
+        }
+        break;
+      default:
+        return;
+    }
+    if (nextRow >= grid.current.rows.length || nextCol >= grid.current.rows[0].cells.length) {
+      event.preventDefault();
+      return;
+    }
+    if (nextCol < 0 || nextRow < 0) {
+      event.preventDefault();
+      return;
+    }
+
+    // Remove the ability to receive visible focus as only one cell should be able to receive tab focus.
+    // This also ensures that tabbing back into the grid focuses on the cell that last had focus.
+    grid.current.rows[currentFocusedRow].cells[currentFocusedCol].tabIndex = -1;
+
+    focusedRow.current = nextRow;
+    focusedCol.current = nextCol;
+
+    const elementWithTabStop = grid.current.rows[nextRow].cells[nextCol];
+    elementWithTabStop.tabIndex = 0;
+    elementWithTabStop.focus();
+    event.preventDefault();
+  };
+
+  const handleClick = (event) => {
+    const clickedCell = event.target.closest('td,th') || document.activeElement.closest('td,th');
+    if (!clickedCell) {
+      event.preventDefault();
+      return;
+    }
+    // Remove Tab stop from previous cell in table that has focus and set it to the cell that was clicked.
+    grid.current.rows[focusedRow.current].cells[focusedCol.current].tabIndex = -1;
+    focusedRow.current = clickedCell.parentElement.rowIndex;
+    focusedCol.current = clickedCell.cellIndex;
+
+    clickedCell.tabIndex = 0;
+    clickedCell.focus();
+    event.preventDefault();
+  };
+
+  const getCellData = (cell, cellColumnIndex) => {
+    const tabIndex = { tabIndex: '-1' };
+    return props.rowHeaderIndex === cellColumnIndex ? (<th key={cellColumnIndex} {...tabIndex} className={cx('worklist-data-grid-row-header')}>{cell.content}</th>) : (<td key={cellColumnIndex} {...tabIndex} className={cx('worklist-data-grid-cell-data')}>{cell.content}</td>);
```

Comment:
Maybe this is for a separate JIRA, but I thought we landed on the worklist does not have row headers.  You can style the cells with a classname without needing to worry about creating a distinction between td and th elements.

Reply:
Our understanding is that the grid will have row headers for screen reader purposes but they will be styled similar to regular cells.